src/R/Components/Impl/Plots/Implementation/PlotClipboardData.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.R.Components.Plots.Implementation {
+    [Serializable]
+    internal class PlotClipboardData {
+        public Guid PlotId { get; }
+        public Guid DeviceId { get; }
+        public int? ProcessId { get; }
+        public bool Cut { get; }
+
+        public const string Format = "RPlotRef";
+
+        public PlotClipboardData(Guid deviceId, Guid plotId, int? processId, bool cut) {
+            DeviceId = deviceId;
+            PlotId = plotId;
+            ProcessId = processId;
+            Cut = cut;
+        }
+
+        // Using .NET serialization wasn't working, error unable to locate the assembly that contained the type
+        // so this is using 'manual' serialization and storing a string type in the clipboard instead.
+        public static PlotClipboardData Parse(string text) {
```

Comment:
Perhaps use Json.NET here to serialize `PlotClipboardData` directly to/from string, and then use that as the clipboard representation?